### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by Paul044**
Task https://epa.ms/nodejs19-hw2

Check task2/README.md

### CODE REVIEW

File: task2/src/routes/user/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    })
+    .put(validators.userWithId, (req, res) => {
+        const { id: userId } = req.params;
+        const storedUser = getUserById(userId);
+        if (!storedUser) {
+            res.status(404).json({
+                message: `User with id ${userId} not found. Update failed`
+            });
+        }
+        if (userId !== storedUser.id) {
```
Comment (meteorra) [inline task2/src/routes/user/index.js]: what is the case when you search user by id & then this user posesses another id rather then the searched one? The only case is the error in your own getUserById function

File: task2/src/routes/user/index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import express from 'express';
+
+import {
+    getUserById,
+    updateUser,
+    deleteUser,
+    createUser
+} from '../../usersData';
```
Comment (meteorra) [inline task2/src/routes/user/index.js:8]: You didn't learn this yet, but all data manipulation functions are usually gathered in services like UserService for example

File: task2/src/routes/user/index.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        const user = req.body;
+        updateUser(user);
+        res.json({
+            message: `User with id ${userId} updated`
+        });
+    })
+    .delete((req, res) => {
+        const { id: userId } = req.params;
+        deleteUser(userId);
```
Comment (meteorra) [inline task2/src/routes/user/index.js]: Validation when user is not found (404)

File: task2/src/routes/user/index.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        res.json({
+            message: `User with id ${userId} deleted`
+        });
+    });
+
+router.put('/', validators.user, (req, res) => {
```
Comment (meteorra) [inline task2/src/routes/user/index.js]: Entity creation semantically is POST request

File: task2/src/index.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+import userRouter from './routes/user';
+import usersRouter from './routes/users';
+
+const app = express();
+const port = process.env.PORT || 3001;
+
+app.listen(port);
+app.use(express.json());
+app.use('/user', userRouter);
```
Comment (meteorra) [inline task2/src/index.js]: Usuall CRUD operations for single entity like user can be created like this:
GET /users/:id - get a user
POST /users - create a user
PUT /users/:id - update a user
DELETE /users/:id - delete a user

For autosuggestion you could use something like /users/autoSuggest or at least /autoSuggest

File: task2/src/routes/user/index.js (hunk 1 above)
Comment (Paul044) [reply task2/src/routes/user/index.js]: removed

File: task2/src/routes/user/index.js (hunk 2 above)
Comment (Paul044) [reply task2/src/routes/user/index.js:8]: Would be done in task3

File: task2/src/routes/user/index.js (hunk 3 above)
Comment (Paul044) [reply task2/src/routes/user/index.js]: Validation added

File: task2/src/routes/user/index.js (hunk 4 above)
Comment (Paul044) [reply task2/src/routes/user/index.js]: changed to POST

File: task2/src/index.js (hunk 5 above)
Comment (Paul044) [reply task2/src/index.js]: Changed to following way

GET /users/:id - get a user
POST /users - create a user
PUT /users/- update a user
DELETE /users/:id - delete a user

/getAutoSuggestUsers

**Pull request merged**